packages/financial-templates-lib/test/price-feed/CoinGeckoPriceFeed.js

```diff
@@ -0,0 +1,226 @@
+const { CoinGeckoPriceFeed } = require("../../src/price-feed/CoinGeckoPriceFeed");
+const { NetworkerMock } = require("../../src/price-feed/NetworkerMock");
+const winston = require("winston");
+
+contract("CoinGeckoPriceFeed.js", function() {
+  let coinGeckoPriceFeed;
+  let networker;
+  let mockTime;
+
+  const contractAddress = "0x6b175474e89094c44da98b954eedeac495271d0f";
+  const currency = "php";
+  const lookback = 120; // 2 minutes.
+  const getTime = () => mockTime;
+  const minTimeBetweenUpdates = 60;
+  const priceFeedDecimals = 18;
+
+  const { toWei, toBN } = web3.utils;
+
+  const mockPrice = 48.15;
+  const validResponse = {
+    [contractAddress]: {
+      [currency]: mockPrice
+    }
+  };
+
+  beforeEach(async function() {
+    networker = new NetworkerMock();
+    mockTime = new Date().getTime();
+
+    const dummyLogger = winston.createLogger({
+      level: "info",
+      transports: [new winston.transports.Console()]
+    });
+
+    coinGeckoPriceFeed = new CoinGeckoPriceFeed(
+      dummyLogger,
+      web3,
+      contractAddress,
+      currency,
+      lookback,
+      networker,
+      getTime,
+      minTimeBetweenUpdates,
+      false,
+      priceFeedDecimals
+    );
+  });
+
+  it("getCurrentPrice() returns the latest price", async function() {
+    networker.getJsonReturns = [validResponse];
+
+    await coinGeckoPriceFeed.update();
+
+    const price = coinGeckoPriceFeed.getCurrentPrice();
+    assert.equal(price.toString(), toWei(`${mockPrice}`));
```

Comment:
Your tests are great and are pretty extensive! However, this set of tests does not validate that the feed can correctly deal with non-18 decimal place precision. could you please add an additional test that double-checks that the feed can handle a non-18 decimals config? I know your product is used dai and as such it should not matter for your synth but for the sake of completeness I'd like to see a test that validates this. 

To do this I think you should do the following in the unit test:
1) create an instance of the `coinGeckoPriceFeed` with `priceFeedDecimals=6` (for USDC, for example)
2) create a decimal conversion object to test the output. see here: https://github.com/UMAprotocol/protocol/blob/71515a915c9d796014295c18ebd6300229f24364/packages/monitors/test/ContractMonitor.js#L86 this basically wraps `parseFixed` in a nice way.
3) execute some of your previously tested logic, such as validating the `getCurrentPrice()` & `getHistoricalPrice()` methods.

Also, please add equivalent tests to the other feed included in this PR.

Reply:
Hi @chrismaree, thanks for the kind words & feedback :)

We have added the unit tests you requested, though 1 note for the `Convert` util - we had to change the code a bit so that it matches the conversion logic used in the price feeds (which we have taken from `CryptoWatchPriceFeed`). Instead of the code above we ended up using:

```
const Convert = decimals => number => parseFixed(number.toString().substring(0, decimals), decimals).toString();
```